src/v2i-hub/SPaTLoggerPlugin/src/SPaTLoggerPlugin.cpp

```diff
@@ -1,236 +1,360 @@
+
+
+	OpenSPaTLogFile();
+	// Add a message filter and handler for each message this plugin wants to receive.
+	//AddMessageFilter<DecodedSPaTMessage>(this, &SPaTLoggerPlugin::HandleDecodedSPaTMessage);
+	AddMessageFilter < SpatMessage > (this, &SPaTLoggerPlugin::HandleSpatMessage);
+
+	// Subscribe to all messages specified by the filters above.
+	SubscribeToMessages();
+
+	PLOG(logDEBUG) << "Exit SPaTLoggerPlugin Constructor";
+}
+
+/**
+ * Destructor
+ */
+
+SPaTLoggerPlugin::~SPaTLoggerPlugin()
+{
+	if (_logFile.is_open())
+	{
+		_logFile.close();
+		_logFilebin.close();
+	}
+}
+
+
+/**
+ * Updates configuration settings
+ */
+void SPaTLoggerPlugin::UpdateConfigSettings()
+{
+	// Configuration settings are retrieved from the API using the GetConfigValue template class.
+	// This method does NOT execute in the main thread, so variables must be protected
+	// (e.g. using std::atomic, std::mutex, etc.).
+
+	int instance;
+	GetConfigValue("Instance", instance);
+
+	GetConfigValue("Frequency", __frequency_mon.get());
+	__frequency_mon.check();
+
+	std::lock_guard<mutex> lock(_cfgLock);
+	GetConfigValue("File Location", _fileDirectory);
+	GetConfigValue("File Size In MB", _maxFilesizeInMB);
+	GetConfigValue("Filename", _filename);
+        std::string oldFilename = _curFilename;
+        std::string oldFilenamebin = _curFilenamebin;
+        _curFilename = _fileDirectory + "/" + _filename + ".json";
+        _curFilenamebin = _fileDirectory + "/" + _filename + ".bin";
+
+        if (_curFilename.compare (oldFilename) !=0 )
+        {
+                _logFile.close();
+                _logFilebin.close();
+                OpenSPaTLogFile();
+        }
+}
+
+/**
+ * Called when configuration is changed
+ *
+ * @param key Key of the configuration value changed
+ * @param value Changed value
+ */
+void SPaTLoggerPlugin::OnConfigChanged(const char *key, const char *value)
+{
+	PluginClient::OnConfigChanged(key, value);
+	UpdateConfigSettings();
+}
+
+/**
+ * Called on plugin state change
+ *
+ * @para state New plugin state
+ */
+void SPaTLoggerPlugin::OnStateChange(IvpPluginState state)
+{
+	PluginClient::OnStateChange(state);
+
+	if (state == IvpPluginState_registered)
+	{
+		UpdateConfigSettings();
+		//SetStatus("ReceivedMaps", 0);
+	}
+}
+
+/**
+ * Method that's called to process a message that this plugin has
+ * subscribed for.  This particular method decodes the SPaT message and
+ * logs selective fields to a log file.
+ *
+ * @param msg SPaTMessage that is received
+ * @routeable_message not used
+ */
+void SPaTLoggerPlugin::HandleSpatMessage(SpatMessage &msg, routeable_message &routeableMsg) 
+{
+
+	char *SpatOut;
+        cJSON *SpatRoot, *SpatMessageContent, *_SpatMessageContent;
+	
+	PLOG(logDEBUG)<<"HandleSPaTMessage";
+	auto spat = msg.get_j2735_data();
+	
+	std::stringstream direction_hex;
+        direction_hex<<"01";
+
+        std::stringstream signStatus_hex;
+        signStatus_hex<<"00";
```

Comment:
Is this supposed to be "verificationStatus" per the updated struct?

Reply:
If yes, please change the variable name to verificationStatus. Or you may import the struct you defined and use the struct variable verificationStatus from there.